{"cells":[{"cell_type":"code","execution_count":1,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":5847,"status":"ok","timestamp":1624440682555,"user":{"displayName":"Vlad Gagarin","photoUrl":"","userId":"01739312221940399891"},"user_tz":-120},"id":"dWS4CIk4NYuw","outputId":"3dd53fa0-00c5-4ddd-900f-5a0052d8c714"},"outputs":[{"name":"stdout","output_type":"stream","text":["\u001b[33mWARNING: Value for scheme.platlib does not match. Please report this to <https://github.com/pypa/pip/issues/10151>\n","distutils: /usr/local/lib/python3.8/dist-packages\n","sysconfig: /usr/lib/python3.8/site-packages\u001b[0m\n","\u001b[33mWARNING: Value for scheme.purelib does not match. Please report this to <https://github.com/pypa/pip/issues/10151>\n","distutils: /usr/local/lib/python3.8/dist-packages\n","sysconfig: /usr/lib/python3.8/site-packages\u001b[0m\n","\u001b[33mWARNING: Value for scheme.headers does not match. Please report this to <https://github.com/pypa/pip/issues/10151>\n","distutils: /usr/local/include/python3.8/UNKNOWN\n","sysconfig: /usr/include/python3.8/UNKNOWN\u001b[0m\n","\u001b[33mWARNING: Value for scheme.scripts does not match. Please report this to <https://github.com/pypa/pip/issues/10151>\n","distutils: /usr/local/bin\n","sysconfig: /usr/bin\u001b[0m\n","\u001b[33mWARNING: Value for scheme.data does not match. Please report this to <https://github.com/pypa/pip/issues/10151>\n","distutils: /usr/local\n","sysconfig: /usr\u001b[0m\n","\u001b[33mWARNING: Additional context:\n","user = False\n","home = None\n","root = None\n","prefix = None\u001b[0m\n","Defaulting to user installation because normal site-packages is not writeable\n","Requirement already satisfied: dnatools in /home/vlad/.local/lib/python3.8/site-packages (0.1)\n","Note: you may need to restart the kernel to use updated packages.\n"]}],"source":["pip install dnatools #Import module "]},{"cell_type":"code","execution_count":2,"metadata":{"executionInfo":{"elapsed":700,"status":"ok","timestamp":1624444172100,"user":{"displayName":"Vlad Gagarin","photoUrl":"","userId":"01739312221940399891"},"user_tz":-120},"id":"YGBB_FypLBpi"},"outputs":[],"source":["from dnatools import * #Package import with DNA tools processing\n","import random #import random for random DNA seq creation\n","import collections"]},{"cell_type":"markdown","metadata":{"id":"BA96rFFmX8wy"},"source":["Kak perwyj szag my dolzny prowerit naszu sekwencju na prawilnost wczitywania, to jest wse li w naszem liste otweczaet normam wpisywania"]},{"cell_type":"code","execution_count":4,"metadata":{"executionInfo":{"elapsed":367,"status":"ok","timestamp":1624444418650,"user":{"displayName":"Vlad Gagarin","photoUrl":"","userId":"01739312221940399891"},"user_tz":-120},"id":"s-V2gUtDNt64"},"outputs":[],"source":["# Check the sequance to make sure it is a DNA String\n","def validateSeq(dna_seq):\n","  tmpseq  = dna_seq\n","  for nuc in tmpseq:#compare all elements in list \n","    if nuc not in nucleotides: #In case DNA seq is not good, will be return bool value\n","      return False\n","  return tmpseq     "]},{"cell_type":"code","execution_count":5,"metadata":{"executionInfo":{"elapsed":415,"status":"ok","timestamp":1624443649159,"user":{"displayName":"Vlad Gagarin","photoUrl":"","userId":"01739312221940399891"},"user_tz":-120},"id":"_TrL-k1xN7G9"},"outputs":[],"source":["nucleotides = ['A','C','G','T'] #test var"]},{"cell_type":"code","execution_count":6,"metadata":{"id":"iKQIrGBONydO"},"outputs":[{"name":"stdout","output_type":"stream","text":["ATTTCGT\n"]}],"source":["print(validateSeq(rndDNASer)) #Validation of DNA seq, in case False on output, DNA seq is not valid"]},{"cell_type":"code","execution_count":7,"metadata":{"executionInfo":{"elapsed":372,"status":"ok","timestamp":1624443679268,"user":{"displayName":"Vlad Gagarin","photoUrl":"","userId":"01739312221940399891"},"user_tz":-120},"id":"v50Ck0EyONgR"},"outputs":[],"source":["randDNAStr = ''.join([random.choice(nucleotides) #lets create our random seq\n","                        for nuc in range(20)])"]},{"cell_type":"code","execution_count":8,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":385,"status":"ok","timestamp":1624444486617,"user":{"displayName":"Vlad Gagarin","photoUrl":"","userId":"01739312221940399891"},"user_tz":-120},"id":"FUO2ogO9VqOK","outputId":"b4a6cca8-68c0-4315-8074-1cdfe5309182"},"outputs":[{"name":"stdout","output_type":"stream","text":["CTATATCATCCCGCGTCACG\n"]}],"source":["print(validateSeq(randDNAStr))"]},{"cell_type":"code","execution_count":9,"metadata":{"executionInfo":{"elapsed":390,"status":"ok","timestamp":1624444431477,"user":{"displayName":"Vlad Gagarin","photoUrl":"","userId":"01739312221940399891"},"user_tz":-120},"id":"1BMOa5vGWxLP"},"outputs":[],"source":["#create a function,which will count nucleotides number in DNA seq\n","def counrNucFrequency(seq): \n","  '''DNA seq nucleotides counter '''\n","  tmpFreqDict = {\"A\":0,\"C\":0,\"G\":0,'T':0}\n","  for nuc in seq:\n","    tmpFreqDict[nuc] += 1\n","  return tmpFreqDict "]},{"cell_type":"code","execution_count":10,"metadata":{"executionInfo":{"elapsed":6,"status":"ok","timestamp":1624444433432,"user":{"displayName":"Vlad Gagarin","photoUrl":"","userId":"01739312221940399891"},"user_tz":-120},"id":"ke_CbgSxYUCk"},"outputs":[],"source":["DNAStr = validateSeq(randDNAStr)"]},{"cell_type":"code","execution_count":11,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":483,"status":"ok","timestamp":1624444497238,"user":{"displayName":"Vlad Gagarin","photoUrl":"","userId":"01739312221940399891"},"user_tz":-120},"id":"3m7-NNIXYQgO","outputId":"f76dd2ae-9b9b-452f-a255-0bc4ed8c0ecb"},"outputs":[{"name":"stdout","output_type":"stream","text":["{'A': 4, 'C': 8, 'G': 3, 'T': 5}\n"]}],"source":["print(counrNucFrequency(DNAStr))"]},{"cell_type":"code","execution_count":12,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":387,"status":"ok","timestamp":1624444443309,"user":{"displayName":"Vlad Gagarin","photoUrl":"","userId":"01739312221940399891"},"user_tz":-120},"id":"bnRdl99yZ8bo","outputId":"d68fabe4-3a30-475d-eaeb-ebc5f3684b3e"},"outputs":[{"name":"stdout","output_type":"stream","text":["{'A': 6, 'C': 2, 'G': 8, 'T': 4}\n"]}],"source":["randDNAStr = ''.join([random.choice(nucleotides) #postrokenie randomnoj sekwencji \n","                        for nuc in range(20)])\n","DNAStr = validateSeq(randDNAStr)\n","print(counrNucFrequency(DNAStr))"]},{"cell_type":"code","execution_count":13,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":384,"status":"ok","timestamp":1624444175285,"user":{"displayName":"Vlad Gagarin","photoUrl":"","userId":"01739312221940399891"},"user_tz":-120},"id":"cyhjp8jTaDRz","outputId":"b88affb3-83f2-4ee5-e547-ed1885dd4145"},"outputs":[{"name":"stdout","output_type":"stream","text":["{'C': 2, 'A': 6, 'G': 8, 'T': 4}\n"]}],"source":["def CounrNucFrequency(seq):\n","  return dict(collections.Counter(seq))\n","print(CounrNucFrequency(DNAStr))"]},{"cell_type":"code","execution_count":14,"metadata":{},"outputs":[],"source":["def trancription(seq):\n","    \"\"\"the function is responsible for transription process, in.oth.h. DNA to RNA\"\"\"\n","    return seq.replace('T','U')"]},{"cell_type":"code","execution_count":15,"metadata":{"id":"DtPhs8ZNawCa"},"outputs":[{"name":"stdout","output_type":"stream","text":["CAGUAUAGCGAUGAGGUGAG\n"]}],"source":["print(trancription(DNAStr))"]},{"cell_type":"code","execution_count":16,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["\n","Sequance : CAGTATAGCGATGAGGTGAG\n","\n","[1]) + Sequance Length: 20\n","\n","[2]) + Nucleotide Frequaency: {'A': 6, 'C': 2, 'G': 8, 'T': 4}\n","\n"]}],"source":["print(f'\\nSequance : {DNAStr}\\n')\n","print(f'[1]) + Sequance Length: {len(DNAStr)}\\n')\n","print(f'[2]) + Nucleotide Frequaency: {counrNucFrequency(DNAStr)}\\n')"]},{"cell_type":"code","execution_count":17,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["[3] + DNA/RNA Transcription : CAGUAUAGCGAUGAGGUGAG\n","\n"]}],"source":["print(f'[3] + DNA/RNA Transcription : {trancription(DNAStr)}\\n')"]},{"cell_type":"code","execution_count":18,"metadata":{},"outputs":[],"source":["DNA_ReverseComplement = {\n","    'A':'T',\n","    'T':'A',\n","    'G':'C',\n","    'C':'G'\n","}"]},{"cell_type":"code","execution_count":19,"metadata":{},"outputs":[],"source":["#Function for complamentetion principo \n","def reverse_complement(seq):\n","    \"\"\"Swaping adenine with thymine and guanine with ctosine. Reversubg newly generated string\"\"\"\n","    return ''.join([DNA_ReverseComplement[nuc] for nuc in seq])[::-1]#[::-1] for reversing of the "]},{"cell_type":"code","execution_count":20,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["tseT\n"]}],"source":["tempStr = 'Test' #reversing our str\n","print(tempStr[::-1]) "]},{"cell_type":"code","execution_count":21,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["CTCACCTCATCGCTATACTG\n"]}],"source":["print(reverse_complement(DNAStr))"]},{"cell_type":"code","execution_count":22,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["[4] + DNA string + Reverse Complement:\n","5'  CAGTATAGCGATGAGGTGAG 3'\n","    ||||||||||||||||||||\n","3'  CTCACCTCATCGCTATACTG 5'\n","\n"]}],"source":["print(f\"[4] + DNA string + Reverse Complement:\\n5'  {DNAStr} 3'\")\n","print(f\"    {''.join(['|' for c in range(len(DNAStr))])}\")\n","print(f\"3'  {reverse_complement(DNAStr)} 5'\\n\")"]},{"cell_type":"code","execution_count":23,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["\n","Sequance : CAGTATAGCGATGAGGTGAG\n","\n","[1]) + Sequance Length: 20\n","\n","[2]) + Nucleotide Frequaency: {'A': 6, 'C': 2, 'G': 8, 'T': 4}\n","\n","[3] + DNA/RNA Transcription : CAGUAUAGCGAUGAGGUGAG\n","\n","[4] + DNA string + Reverse Complement:\n","5'  CAGTATAGCGATGAGGTGAG 3'\n","    ||||||||||||||||||||\n","3'  CTCACCTCATCGCTATACTG 5'\n","\n"]}],"source":["print(f'\\nSequance : {DNAStr}\\n')\n","print(f'[1]) + Sequance Length: {len(DNAStr)}\\n')\n","print(f'[2]) + Nucleotide Frequaency: {counrNucFrequency(DNAStr)}\\n')\n","print(f'[3] + DNA/RNA Transcription : {trancription(DNAStr)}\\n')\n","print(f\"[4] + DNA string + Reverse Complement:\\n5'  {DNAStr} 3'\")\n","print(f\"    {''.join(['|' for c in range(len(DNAStr))])}\")\n","print(f\"3'  {reverse_complement(DNAStr)} 5'\\n\")"]},{"cell_type":"code","execution_count":24,"metadata":{},"outputs":[],"source":["from struct import *"]},{"cell_type":"code","execution_count":25,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["\n","Sequance : CAGTATAGCGATGAGGTGAG\n","\n","[1]) + Sequance Length: 20\n","\n","[2]) + Nucleotide Frequaency: {'A': 6, 'C': 2, 'G': 8, 'T': 4}\n","\n","[3] + DNA/RNA Transcription : CAGUAUAGCGAUGAGGUGAG\n","\n","[4] + DNA string + Reverse Complement:\n","5'  CAGTATAGCGATGAGGTGAG   3'\n","    ||||||||||||||||||||\n","3'  CTCACCTCATCGCTATACTG   5'\n","\n"]}],"source":["print(f'\\nSequance : {DNAStr}\\n')\n","print(f'[1]) + Sequance Length: {len(DNAStr)}\\n')\n","print(f'[2]) + Nucleotide Frequaency: {counrNucFrequency(DNAStr)}\\n')\n","print(f'[3] + DNA/RNA Transcription : {trancription(DNAStr)}\\n')\n","print(f\"[4] + DNA string + Reverse Complement:\\n5'  {DNAStr}   3'\")\n","print(f\"    {''.join(['|' for c in range(len(DNAStr))])}\")\n","print(f\"3'  {reverse_complement(DNAStr)}   5'\\n\")"]},{"cell_type":"code","execution_count":26,"metadata":{},"outputs":[],"source":["nucleotides = ['A','C','G','T']\n","DNA_ReverseComplement = {\n","    'A':'T',\n","    'T':'A',\n","    'G':'C',\n","    'C':'G'\n","}"]},{"cell_type":"code","execution_count":27,"metadata":{},"outputs":[],"source":["#utilites \n","def colored(seq):\n","    bcolors = {\n","        \"A\": '\\033[92m',\n","        'C': '\\033[94m',\n","        'G': '\\033[93m',\n","        'T': '\\033[91m',\n","        'U': '\\033[91m',\n","        'reset': '\\033[0;0m'        \n","    }\n","    tmpStr = \"\"\n","    for nuc in seq:\n","        if nuc in bcolors:\n","            tmpStr +=bcolors[nuc] + nuc\n","        else:\n","            tmpStr +=bcolors['reset'] + nuc\n","    return tmpStr + '\\033[0;0m'  "]},{"cell_type":"code","execution_count":28,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["\n","Sequance : \u001b[94mC\u001b[92mA\u001b[93mG\u001b[91mT\u001b[92mA\u001b[91mT\u001b[92mA\u001b[93mG\u001b[94mC\u001b[93mG\u001b[92mA\u001b[91mT\u001b[93mG\u001b[92mA\u001b[93mG\u001b[93mG\u001b[91mT\u001b[93mG\u001b[92mA\u001b[93mG\u001b[0;0m\n","\n","[1]) + Sequance Length: 20\n","\n","[2]) + Nucleotide Frequaency: \u001b[92mA\u001b[94mC\u001b[93mG\u001b[91mT\u001b[0;0m\n","\n","[3] + DNA/RNA Transcription : \u001b[94mC\u001b[92mA\u001b[93mG\u001b[91mU\u001b[92mA\u001b[91mU\u001b[92mA\u001b[93mG\u001b[94mC\u001b[93mG\u001b[92mA\u001b[91mU\u001b[93mG\u001b[92mA\u001b[93mG\u001b[93mG\u001b[91mU\u001b[93mG\u001b[92mA\u001b[93mG\u001b[0;0m\n","\n","[4] + DNA string + Reverse Complement:\n","5'  \u001b[94mC\u001b[92mA\u001b[93mG\u001b[91mT\u001b[92mA\u001b[91mT\u001b[92mA\u001b[93mG\u001b[94mC\u001b[93mG\u001b[92mA\u001b[91mT\u001b[93mG\u001b[92mA\u001b[93mG\u001b[93mG\u001b[91mT\u001b[93mG\u001b[92mA\u001b[93mG\u001b[0;0m   3'\n","    ||||||||||||||||||||\n","3'  \u001b[94mC\u001b[91mT\u001b[94mC\u001b[92mA\u001b[94mC\u001b[94mC\u001b[91mT\u001b[94mC\u001b[92mA\u001b[91mT\u001b[94mC\u001b[93mG\u001b[94mC\u001b[91mT\u001b[92mA\u001b[91mT\u001b[92mA\u001b[94mC\u001b[91mT\u001b[93mG\u001b[0;0m   5'\n","\n"]}],"source":["print(f'\\nSequance : {colored(DNAStr)}\\n')\n","print(f'[1]) + Sequance Length: {len(DNAStr)}\\n')\n","print(f'[2]) + Nucleotide Frequaency: {colored(counrNucFrequency(DNAStr))}\\n')\n","print(f'[3] + DNA/RNA Transcription : {colored(trancription(DNAStr))}\\n')\n","print(f\"[4] + DNA string + Reverse Complement:\\n5'  {colored(DNAStr)}   3'\")\n","print(f\"    {''.join(['|' for c in range(len(DNAStr))])}\")\n","print(f\"3'  {colored(reverse_complement(DNAStr))}   5'\\n\")"]},{"cell_type":"code","execution_count":29,"metadata":{},"outputs":[],"source":["def gc_content(seq):\n","    \"\"\"GC content in DNA/RNA sequance\"\"\"\n","    return round((seq.count('C')+seq.count('G'))/len(seq)*100)"]},{"cell_type":"code","execution_count":30,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["50\n"]}],"source":["print(gc_content(DNAStr))"]},{"cell_type":"code","execution_count":31,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["[5] +GC content: 50 % \n","\n"]}],"source":["print(f'[5] +GC content: {gc_content(DNAStr)} % \\n')"]},{"cell_type":"code","execution_count":32,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["\n","Sequance : \u001b[94mC\u001b[92mA\u001b[93mG\u001b[91mT\u001b[92mA\u001b[91mT\u001b[92mA\u001b[93mG\u001b[94mC\u001b[93mG\u001b[92mA\u001b[91mT\u001b[93mG\u001b[92mA\u001b[93mG\u001b[93mG\u001b[91mT\u001b[93mG\u001b[92mA\u001b[93mG\u001b[0;0m\n","\n","[1]) + Sequance Length: 20\n","\n","[2]) + Nucleotide Frequaency: \u001b[92mA\u001b[94mC\u001b[93mG\u001b[91mT\u001b[0;0m\n","\n","[3] + DNA/RNA Transcription : \u001b[94mC\u001b[92mA\u001b[93mG\u001b[91mU\u001b[92mA\u001b[91mU\u001b[92mA\u001b[93mG\u001b[94mC\u001b[93mG\u001b[92mA\u001b[91mU\u001b[93mG\u001b[92mA\u001b[93mG\u001b[93mG\u001b[91mU\u001b[93mG\u001b[92mA\u001b[93mG\u001b[0;0m\n","\n","[4] + DNA string + Reverse Complement:\n","5'  \u001b[94mC\u001b[92mA\u001b[93mG\u001b[91mT\u001b[92mA\u001b[91mT\u001b[92mA\u001b[93mG\u001b[94mC\u001b[93mG\u001b[92mA\u001b[91mT\u001b[93mG\u001b[92mA\u001b[93mG\u001b[93mG\u001b[91mT\u001b[93mG\u001b[92mA\u001b[93mG\u001b[0;0m   3'\n","    ||||||||||||||||||||\n","3'  \u001b[94mC\u001b[91mT\u001b[94mC\u001b[92mA\u001b[94mC\u001b[94mC\u001b[91mT\u001b[94mC\u001b[92mA\u001b[91mT\u001b[94mC\u001b[93mG\u001b[94mC\u001b[91mT\u001b[92mA\u001b[91mT\u001b[92mA\u001b[94mC\u001b[91mT\u001b[93mG\u001b[0;0m   5'\n","\n","[5] + GC content: 50 % \n","\n"]}],"source":["print(f'\\nSequance : {colored(DNAStr)}\\n')\n","print(f'[1]) + Sequance Length: {len(DNAStr)}\\n')\n","print(f'[2]) + Nucleotide Frequaency: {colored(counrNucFrequency(DNAStr))}\\n')\n","print(f'[3] + DNA/RNA Transcription : {colored(trancription(DNAStr))}\\n')\n","print(f\"[4] + DNA string + Reverse Complement:\\n5'  {colored(DNAStr)}   3'\")\n","print(f\"    {''.join(['|' for c in range(len(DNAStr))])}\")\n","print(f\"3'  {colored(reverse_complement(DNAStr))}   5'\\n\")\n","print(f'[5] + GC content: {gc_content(DNAStr)} % \\n')"]},{"cell_type":"code","execution_count":33,"metadata":{},"outputs":[],"source":["def gc_content_subsec(seq, k =20):\n","    \"\"\"GC content in a DNA/RNA sub-sequance lenght k. K = 20 by defolt\"\"\"\n","    res = []\n","    for i in range(0,len(seq) - k + 1, k):\n","        subseq = seq[i:i + k]\n","        res.append(gc_content(subseq))\n","    return res"]},{"cell_type":"code","execution_count":34,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["[6] + GC Content in Subsection k=5: [40, 60, 40, 60]\n","\n"]}],"source":["print(f'[6] + GC Content in Subsection k=5: {gc_content_subsec(DNAStr,k =5)}\\n')"]},{"cell_type":"code","execution_count":35,"metadata":{},"outputs":[],"source":["# from collections import Counter\n","from collections import Counter"]},{"cell_type":"code","execution_count":36,"metadata":{},"outputs":[],"source":["def translate_seq(seq, init_pos=0):\n","    \"\"\"Translates a DNA sequence into an aminoacid sequence\"\"\"\n","    return [DNA_Codons[seq[pos:pos + 3]] for pos in range(init_pos, len(seq) - 2, 3)]"]},{"cell_type":"code","execution_count":37,"metadata":{},"outputs":[],"source":["def codon_usage(seq, aminoacid):\n","    \"\"\"Provides the frequency of each codon encoding a given aminoacid in a DNA sequence\"\"\"\n","    tmpList = []\n","    for i in range(0, len(seq) - 2, 3):\n","        if DNA_Codons[seq[i:i + 3]] == aminoacid:\n","            tmpList.append(seq[i:i + 3])\n","\n","    freqDict = dict(Counter(tmpList))\n","    totalWight = sum(freqDict.values())\n","    for seq in freqDict:\n","        freqDict[seq] = round(freqDict[seq] / totalWight, 2)\n","    return freqDict"]},{"cell_type":"code","execution_count":38,"metadata":{},"outputs":[],"source":["DNA_Codons = {\n","    # 'M' - START, '_' - STOP\n","    \"GCT\": \"A\", \"GCC\": \"A\", \"GCA\": \"A\", \"GCG\": \"A\",\n","    \"TGT\": \"C\", \"TGC\": \"C\",\n","    \"GAT\": \"D\", \"GAC\": \"D\",\n","    \"GAA\": \"E\", \"GAG\": \"E\",\n","    \"TTT\": \"F\", \"TTC\": \"F\",\n","    \"GGT\": \"G\", \"GGC\": \"G\", \"GGA\": \"G\", \"GGG\": \"G\",\n","    \"CAT\": \"H\", \"CAC\": \"H\",\n","    \"ATA\": \"I\", \"ATT\": \"I\", \"ATC\": \"I\",\n","    \"AAA\": \"K\", \"AAG\": \"K\",\n","    \"TTA\": \"L\", \"TTG\": \"L\", \"CTT\": \"L\", \"CTC\": \"L\", \"CTA\": \"L\", \"CTG\": \"L\",\n","    \"ATG\": \"M\",\n","    \"AAT\": \"N\", \"AAC\": \"N\",\n","    \"CCT\": \"P\", \"CCC\": \"P\", \"CCA\": \"P\", \"CCG\": \"P\",\n","    \"CAA\": \"Q\", \"CAG\": \"Q\",\n","    \"CGT\": \"R\", \"CGC\": \"R\", \"CGA\": \"R\", \"CGG\": \"R\", \"AGA\": \"R\", \"AGG\": \"R\",\n","    \"TCT\": \"S\", \"TCC\": \"S\", \"TCA\": \"S\", \"TCG\": \"S\", \"AGT\": \"S\", \"AGC\": \"S\",\n","    \"ACT\": \"T\", \"ACC\": \"T\", \"ACA\": \"T\", \"ACG\": \"T\",\n","    \"GTT\": \"V\", \"GTC\": \"V\", \"GTA\": \"V\", \"GTG\": \"V\",\n","    \"TGG\": \"W\",\n","    \"TAT\": \"Y\", \"TAC\": \"Y\",\n","    \"TAA\": \"_\", \"TAG\": \"_\", \"TGA\": \"_\"\n","}"]},{"cell_type":"code","execution_count":39,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["[5] + GC Content: 50%\n","\n","[6] + GC Content in Subsection k=5: [40, 60, 40, 60]\n","\n","[7] + Aminoacids Sequence from DNA: ['Q', 'Y', 'S', 'D', 'E', 'V']\n","\n","[8] + Codon frequency (L): {}\n","\n"]}],"source":["print(f'[5] + GC Content: {gc_content(DNAStr)}%\\n')\n","print(\n","    f'[6] + GC Content in Subsection k=5: {gc_content_subsec(DNAStr, k=5)}\\n')\n","\n","print(\n","    f'[7] + Aminoacids Sequence from DNA: {translate_seq(DNAStr, 0)}\\n')\n","\n","print(\n","    f'[8] + Codon frequency (L): {codon_usage(DNAStr, \"L\")}\\n')\n"]},{"cell_type":"code","execution_count":40,"metadata":{},"outputs":[],"source":["from Bio.Seq import Seq\n","from Bio.SeqUtils import GC\n","import pandas as pd\n","def readFile(filePath):\n","    '''Reading the file and ruturn list of lines'''\n","    with open(filePath,'r') as f:\n","        return[l.strip() for l in f.readlines()]\n","\n","def gc_content(seq):\n","    '''Retur GC content in aa DNA/RNA sequance'''\n","    return ((seq.count('C')+seq.count('G')/len(seq)*100))"]},{"cell_type":"code","execution_count":53,"metadata":{},"outputs":[],"source":["#=== Clean/Prepara the data (Format for ease of you with our GC_content func)\n","#Converting FASTA/List file data into a dictionary\n","for line in FASTAFile:\n","    if '@' in line:\n","        FASTALabel = line\n","        FASTADict[FASTALabel] = ''\n","    else:\n","        FASTADict[FASTALabel] += line \n","\n","for line in FASTADict:\n","    if '+' in FASTADict[()]:"]}],"metadata":{"colab":{"authorship_tag":"ABX9TyNhjQxm8YMBZS2Hc6wLj2Dc","collapsed_sections":[],"name":"BioPy(trial).ipynb","provenance":[]},"interpreter":{"hash":"916dbcbb3f70747c44a77c7bcd40155683ae19c65e1c03b4aa3499c5328201f1"},"kernelspec":{"display_name":"Python 3.8.10 64-bit","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.8.10"}},"nbformat":4,"nbformat_minor":2}
